Tuesdays/Week-Class/Inheratance/Assets/Scripts/PlayerController.cs

```diff
@@ -9,11 +9,16 @@ public class PlayerController : MonoBehaviour{
 	[SerializeField] private Rigidbody2D _rb2D;
 	int num;
 	PlayableCharacter currCharacter;
-
-	public PlayableCharacter CurrCharacter { get { return currCharacter; } }
+	private Stack<Vector3> _playerMovementHistory;
 
 
+	public PlayableCharacter CurrCharacter { get { return currCharacter; } }
+	public Stack<Vector3> PlayerMovementHistory { get { return _playerMovementHistory; } }
```

Comment:
you can use [Expression-bodied member](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/statements-expressions-operators/expression-bodied-members).
( `=>` if you ever saw it)
and transform the stack to be read-only to external classes
it will look like this 
```cs
public Stack<Vector3> PlayerMovementHistory => _playerMovementHistory;
```

Reply:
but as you can understand, you violate the point of making the stack private, as you are exposing the instance to every one.
it would be better to keep the stack private and expose only the the exact functions you need that use the stack.

plus that way if you ever change the stack to something else, you will not have a breaking change